### DISCUSSION TITLE
Why some library in some triplets depends on the other triplet, such as ICU and flatbuffers?

### DISCUSSION THREAD

**Discussion by Colysia** [microsoft/vcpkg#39980] [Q&A]
As described in my title, when I install `icu:x64-windows-static-md` or `icu:x64-mingw-static`, either depends on `icu:x64-windows`, as well as flatbuffers, protobuf, and sqlite3. And I try to remove its dependence on itself, but cannot build successfully. Could someone tell me what considerations this is for?

**Comment by dg0yt** [comment]
vcpkg ports are only allowed to build binaries for the target triplet. If a port needs binaries to be executed on the host, they must be used via host dependencies. If the binary is provided by the package itself, it is modeled by a host dependency on the port itself.

If your host system and your target triplet match well enough, you may simply set the host triplet to the target triplet.
